### DISCUSSION TITLE
Call or Forward?

### DISCUSSION THREAD

**Discussion by keunwoochoi** [Lightning-AI/pytorch-lightning#5811] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi, thanks for the nice library. In the `readme`, the example uses  `model.forward(x)` not `model(x)`. But wouldn't it usually recommended to use `model(x)` so that other things (hooks etc) can be, well, hooked as well? What's the best practice?

**Comment by williamFalcon** [comment, marked as answer]
forward should implement what you want to use when calling model(x). 

you may need to call that in training step (usually do), which means you have to do self.forward(...) because you are in the model when you make that call.
